docs/testplan/transceiver_onboarding_test_plan.md

```diff
@@ -221,14 +267,188 @@ The following tests aim to validate various functionalities of the transceiver u
 | Verify transceiver error-status | Validate CLI relying on redis-db | Ensure the relevant port is in an "OK" state |
 | Verify transceiver error-status with hardware verification | Validate CLI relying on transceiver hardware | Ensure the relevant port is in an "OK" state |
 
-#### 1.4 Firmware Related Tests
+#### 1.4 CMIS CDB Firmware Upgrade Testing
 
-**Pre-requisite**
+##### 1.4.1 CMIS CDB Firmware Binary Management
 
-1. DOM polling must be disabled to prevent race conditions between I2C transactions and the CDB mode for modules that cannot support CDB background mode.
-2. On some platforms, `thermalctld` or a similar user process that performs I2C transactions with the module may need to be stopped.
-3. Two gold firmware versions (A and B) are required so that the system can switch between them multiple times (assuming both versions support the CDB protocol).
-4. The module must support dual banks.
+###### 1.4.1.1 Firmware Binary Naming Guidelines
+
+CMIS CDB firmware binaries **must** not have not have spaces or special characters (except the below special characters) in their filenames.
+The valid special characters are `-` and `_`.
+
+###### 1.4.1.2 Normalization Rules for Vendor Name and Part Number
+
+To ensure compatibility and uniqueness across filesystems and automation tools, the following normalization rules should be applied to vendor names and part numbers:
+
+> **Important Note:** These normalization rules are designed for test framework consumption and firmware binary storage organization. They are **not** applied to the actual transceiver EEPROM data, which remains unchanged.
+
+**Core Normalization Rules:**
+
+1. **Character Replacement:**
+   - Preserve hyphens (`-`) and underscores (`_`) as they are filesystem-safe
+   - Replace all other non-alphanumeric characters (spaces, `/`, `.`, `&`, `#`, `@`, `%`, `+`, etc.) with underscores (`_`)
```

Comment:
@mihirpat1 this means Vendors cannot have these non-alphanumeric characters? if they do have, we will replace them with _?